staging/src/k8s.io/apiserver/pkg/registry/generic/registry/corrupt_obj_deleter.go

```diff
@@ -101,6 +105,11 @@ func (d *corruptObjectDeleter) Delete(ctx context.Context, name string, deleteVa
 		return obj, deleted, err
 	}
 
+	if dryrun.IsDryRun(opts.DryRun) {
```

Comment:
shouldn't we put it before the normal deletion? In case https://github.com/kubernetes/kubernetes/pull/134037/files#diff-592aa733e64bdba1062e9957a7d5b0f4124e7f8102ecb4e4278c7d8215e21af6R98 this passes.

Also not sure but can this pass in any case?

Reply:
I added a comment in line 97  that highlights the fact that the `d.store` is "classic DRY-RUN". Maybe I could update the comment to highlight that this is the [DryRunnableStorage](https://github.com/kubernetes/kubernetes/blob/aa35105213d031c25abf172ac1102a58f144c85e/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L221).

So it should be able to handle the default DRY-RUN out of the box.

Reply:
yes, thanks for sharing but as I see the code now here: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L1200 

which is doing something similar with an extra check on output.